backend/internal/database/book.go

```diff
@@ -91,3 +94,161 @@ func (db *BookDatabase) FetchBookDetails(ctx context.Context,
 
 	return &book, false, nil
 }
+
+/*
+ * Fetches a book entry from the database given the id
+ */
+func (db *BookDatabase) FetchBook(ctx context.Context, id string) (*models.Book, error) {
+	var book models.Book
+	var query = "SELECT books.id, title, author, image, array_remove(array_agg(lang), NULL) as languages," +
+		"created_at FROM books LEFT JOIN book_contents ON " +
+		"books.id = book_contents.id WHERE books.id = $1 GROUP BY books.id"
+
+	err := db.Conn.QueryRowEx(ctx, query, nil, id).Scan(&book.ID, &book.Title, &book.Author, &book.Image,
+		&book.Languages, &book.CreatedAt)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "error on FetchBook")
+	}
+
+	return &book, nil
+}
+
+/*
+ * Inserts a book into the books table
+ * MUST be called on the first instance a book is inserted into the database
+ * and has no associated language
```

Comment:
Is this because of the `[NULL]` bug from before? If so I don't think this is true anymore